components/src/dynamo/planner/tests/unit/test_engine_query.py

```
@@ -126,6 +134,314 @@ def best_available(cls, config, options):
+def test_native_correction_cap_fails_closed_when_baseline_estimate_is_unavailable():
+    model = AicCoreEnginePerfModel(
+        model=_FakeForwardPassModel(lambda _metrics: 80.0),
+        uncorrected_native_model=_FakeForwardPassModel(lambda _metrics: None),
+        max_correction_factor=2.0,
+        worker_type="decode",
+        limits=EnginePerfLimits(128, 16, 10_000),
+        attention_dp_size=1,
+        max_observations=8,
+    )
+
+    with pytest.raises(ValueError, match="native AIC.*unavailable"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)